src/app/components/plugin-sidebar/plugin-sidebar.component.ts

```diff
@@ -101,6 +101,46 @@ export class PluginSidebarComponent implements OnInit, OnDestroy {
                 this.switchActiveTemplateLink(template?.self ?? null);
             }
         });
+        this.registry.newApiObjectSubject
+            .pipe(filter(newObject => newObject.new.resourceType === "ui-template-tab"))
+            .subscribe(async newObject => {
```

Comment:
I generally try to move larger lambdas out of the subscribe functions into their own methods. They get a name this way and my observable pipes are shorter and more readable (especially if there are multiple pipes/subscriptions in the same method. (One exception is the route params subscription, which I also mostly handle inline in `onInit`.